docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,264 @@
+We take advantage of this asymmetry and explicitly label one of the chains with `I` (the initiating, or source, process), and the other with `R` (the responder, or destination, process).
+
+On top of each chain, one or multiple **applications** may be running (e.g., payment system or token exchanges, with their associated smart contracts).
+In the context of the connection handshake problem, it is an application that invokes the connection handshake protocol, and provides the necessary input for executing this protocol (we give more details on the invocation & input in section [2.2 Protocol](#22-protocol) below).
+
+At a high-level, the handshake protocol consists of processes `I` and `R` exchanging a few messages called **datagrams**.
+Each datagram has a well-defined syntax which we describe later.
+__[FIX: describe this and put ref to [2.2 Protocol](#22-protocol).]__
+
+We define a **connection** as a bidirectional or unidirectional path of communication that applications on the initiator and responder chains use to exchange **packets**.
+Packets are payloads from the point of view of a connection; they contain arbitrary application-level data, and their content is opaque to a connection; suffice to say that applications generate and consume packets.
+To summarize, the purpose of a connection is to enable chains `I` and `R` -- as well as applications running on top of these chains -- to coordinate updates to their state.
+
+
+###### Communication assumptions
+We assume that process `I` and `R` have access to an underlying communication layer (e.g., message passing over TCP/IP) and each process has connectivity with the other.
+
+The code snippet below describes the communication layer API.
+This API comprises a function `Send` for invoking the communication layer to send a datagram.
+And there is a callback `Deliver` that each process executes upon delivering a datagram from the communication layer.
+
+```golang
+type CommunicationLayer interface {
+    // A process `s` invokes this function to send a datagram `g` to
+    // a destination process `d`.
+    Send(g Datagram, d Process) null
+
+    // This callback triggers at process `d` to deliver datagram `g`
+    // that process `s` previously sent.
+    Deliver(g Datagram, s Process) null
+}
+```
+
+Neither of these functions are blocking, and they have no return value.
+We also assume that datagrams cannot be tampered with, i.e., they are authenticated by the sender process when calling `Send`.
+We assume that the destination process implicitly verifies authenticity upon calling `Deliver` for any datagram.
+
+The communication layer is _asynchronous_.
+There is no guarantee on how much time it will take for the destination of a datagram to deliver that datagram.
+
+
+###### Fault assumptions
+This specification operates under the assumption that processes `I` and `R` are not faulty, i.e., do not deviate from any protocol we describe below.
```

Comment:
My understanding is that we operate in the model where processes can behave arbitrarily (Byzantine fault), but guarantees are given only for correct case.

Reply:
Hmm, I think I was vague here. Maybe it would help to restate this as follows? "Processes `i` and `r` may be subject to arbitrary/Byzantine faults, but the properties we define below only hold as long as both `i` and `r` behave correctly.